src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -4856,10 +4863,11 @@ private bool RemoteDestinationPathIsFile(string destination, System.Management.A
             return (bool)(op["IsFileInfo"]);
         }
 
-        private string CreateDirectoryOnRemoteSession(string destination, bool force, System.Management.Automation.PowerShell ps)
+        private string CreateDirectoryOnRemoteSession(DirectoryInfo sourceDirectory, string destination, bool force, System.Management.Automation.PowerShell ps)
         {
             ps.AddCommand(CopyFileRemoteUtils.PSCopyToSessionHelperName);
             ps.AddParameter("createDirectoryPath", destination);
+            ps.AddParameter("attributes", sourceDirectory.Attributes);
```

Comment:
To put this feature under experimental, you will have to guard this line in a `if (ExperimentalFeature.IsEnabled(....))`.